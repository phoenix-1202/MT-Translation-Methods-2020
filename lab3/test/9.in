mygcd :: Integer -> Integer -> Integer
mygcd a b | a == 0 || b == 0 = a + b
mygcd a b = if a < b then (mygcd a (b `mod` a)) else (mygcd (a `mod` b) b)

nokWithOperationsOrderTroubles :: Integer -> Integer -> Integer
nokWithOperationsOrderTroubles a b | (mygcd a b) == 1 = a * b
nokWithOperationsOrderTroubles a b = a * b / (mygcd a b)