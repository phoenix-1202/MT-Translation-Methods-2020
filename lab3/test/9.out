#include <bits/stdc++.h>;
using namespace std;

long long mygcd(long long arg1, long long arg2) {
if ((arg1 == 0) || (arg2 == 0)) {
return (arg1 + arg2);
}
if ((arg1 < arg2)) {
return mygcd(arg1, (arg2 % arg1));
}
else {
return mygcd((arg1 % arg2), arg2);
}
}

long long nokWithOperationsOrderTroubles(long long arg1, long long arg2) {
if ((mygcd(arg1, arg2) == 1)) {
return (arg1 * arg2);
}
return (arg1 * (arg2 / mygcd(arg1, arg2)));
}

//EXIT